src/otx/core/data/entity/classification.py

```diff
@@ -144,3 +146,210 @@ def pin_memory(self) -> MultilabelClsBatchDataEntity:
+
+            def put_key_values(src: dict, dst: dict) -> None:
+                """Put key and values from src to dst."""
+                for k, v in src.items():
+                    dst[k] = v
+
+            class_to_idx: dict[str, tuple[int, int]] = {}
+            put_key_values(exclusive_ctoi, class_to_idx)
+            put_key_values(single_label_ctoi, class_to_idx)
+            return class_to_idx
+
+        all_groups = [label_group.labels for label_group in dm_label_categories.label_groups]
+
+        exclusive_group_info = get_exclusive_group_info(all_groups)
+        single_label_group_info = get_single_label_group_info(all_groups, exclusive_group_info["num_multiclass_heads"])
+
+        merged_class_to_idx = merge_class_to_idx(
+            exclusive_group_info["class_to_idx"],
+            single_label_group_info["class_to_idx"],
+        )
+
+        return HLabelInfo(
+            num_multiclass_heads=exclusive_group_info["num_multiclass_heads"],
+            num_multilabel_classes=single_label_group_info["num_multilabel_classes"],
+            head_idx_to_logits_range=exclusive_group_info["head_idx_to_logits_range"],
+            num_single_label_classes=exclusive_group_info["num_single_label_classes"],
+            class_to_group_idx=merged_class_to_idx,
+            all_groups=all_groups,
+            label_to_idx=dm_label_categories._indices,  # noqa: SLF001
+            empty_multiclass_head_indices=[],  # consider the label removing case
+        )
+
+
+@register_pytree_node
+@dataclass
+class HlabelClsDataEntity(OTXDataEntity):
+    """Data entity for H-label classification task.
+
+    :param labels: labels as integer indices
+    :param label_group: the group of the label
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.H_LABEL_CLS
+
+    labels: LongTensor
+
+
+@dataclass
+class HlabelClsPredEntity(HlabelClsDataEntity, OTXPredEntity):
+    """Data entity to represent the H-label classification model output prediction."""
+
+
+@dataclass
+class HlabelClsBatchDataEntity(OTXBatchDataEntity[HlabelClsDataEntity]):
+    """Data entity for H-label classification task.
+
+    :param labels: A list of labels as integer indices
+    :param label_groups: A list of label group
+    """
+
+    labels: list[LongTensor]
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.H_LABEL_CLS
+
+    @classmethod
+    def collate_fn(
+        cls,
+        entities: list[HlabelClsDataEntity],
+    ) -> HlabelClsBatchDataEntity:
+        """Collection function to collect `OTXDataEntity` into `OTXBatchDataEntity` in data loader."""
+        batch_data = super().collate_fn(entities)
+        return HlabelClsBatchDataEntity(
+            batch_size=batch_data.batch_size,
+            images=batch_data.images,
```

Comment:
Please use the batch-wise stacked image tensor such as
https://github.com/openvinotoolkit/training_extensions/blob/fdc033ec6b1fdc6615705ddd75af960a49af9f1d/src/otx/core/data/entity/classification.py#L134

Reply:
Done, https://github.com/openvinotoolkit/training_extensions/pull/2761/commits/d2fd2089fe44c8c81a573760634ea0992750fa48